prottp.go

```diff
@@ -151,39 +185,41 @@ func wrapMethod(service interface{}, m grpc.MethodDesc) http.Handler {
 			}
 			return
 		}
-		writeMessage(isJson, resp.(proto.Message), http.StatusOK, w)
+		WriteMessage(statusCode, resp.(proto.Message), w, r)
 	})
 }
 
-func writeMessage(isJson bool, msg proto.Message, statusCode int, w http.ResponseWriter) {
+// WriteMessage is exported to be used for middleware to return proto message
+func WriteMessage(statusCode int, msg proto.Message, w http.ResponseWriter, r *http.Request) {
 	var err error
+	var isJSON = isMimeTypeJSON(w.Header().Get("Content-Type"))
+	if w.Header().Get("Content-Type") == "" {
+		isJSON = shouldReturnJSON(r)
+		if isJSON {
+			w.Header().Set("Content-Type", fmt.Sprintf("%s;type=%s", jsonContentType, proto.MessageName(msg)))
+		} else {
+			w.Header().Set("Content-Type", fmt.Sprintf("%s;type=%s", xprottpContentType, proto.MessageName(msg)))
+		}
+	}
 
-	// if msg == nil || reflect.ValueOf(msg).IsNil() {
-	// 	if statusCode == http.StatusOK {
-	// 		w.WriteHeader(http.StatusNoContent)
-	// 	}
-	// 	return
-	// }
+	if statusCode > 0 {
+		w.WriteHeader(statusCode)
+	}
 
-	if isJson {
+	// start write body
+	if isJSON {
```

Comment:
Is there any reason why this check of `isJSON` can't be unified with the check above?

Reply:
Can's start writing body, with out first `w.WriteHeader` write headers.